meson.build

```diff
@@ -139,15 +138,23 @@ else
 endif
 
 
-# XXX - why do we have to define this manually?
-if (cxx.has_header('libintl.h'))
-  add_project_arguments('-DHAVE_GETTEXT', language: ['c', 'cpp'])
-
-  if have_darwin or have_freebsd or have_windows
-    add_project_link_arguments('-lintl', language: ['c', 'cpp'])
+if meson.version().version_compare('>= 0.59')
+  intl_dep = dependency('intl', required: false)
+  intl_found = intl_dep.found()
+else
+  if cxx.has_function('ngettext')
```

Comment:
See https://github.com/audacious-media-player/audacious/pull/62#discussion_r1167998267, this needs changes to work.

Reply:
In fact, that's where your CI error comes from:
```
D:/a/_temp/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: src/metronom/metronom.dll.p/metronom.cc.obj: in function `dgettext':

D:/a/_temp/msys64/mingw32/include/libintl.h:150: undefined reference to `libintl_dgettext'
```